src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForFluentDiagnosticAnalyzer.cs

```diff
@@ -187,7 +199,7 @@ private static bool AnalyzeInvocation(
             // left hand side of the expression.  In the inner expressions we can have things like `.Concat/.Append`
             // calls as the outer expressions will realize the collection.
             if (current is InvocationExpressionSyntax { Expression: MemberAccessExpressionSyntax currentMemberAccess } currentInvocation &&
-                IsMatch(state, currentMemberAccess, currentInvocation, allowLinq: true, matchesInReverse, out _, cancellationToken))
+                IsMatch(state, currentMemberAccess, currentInvocation, allowLinq: true, postMatchesInReverse, out _, cancellationToken))
```

Comment:
>IsMatch [](http://example.com/codeflow?start=16&length=7)

I think this would be easier for me to read if the while loop ended after this if block

Reply:
will make that chnage.

Reply:
this got much nicer.  thanks.